homeassistant/components/sensor/insteon_plm.py

```diff
@@ -0,0 +1,95 @@
+"""
+Support for INSTEON dimmers via PowerLinc Modem.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/insteon_plm/
+"""
+import logging
+import asyncio
+
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.loader import get_component
+
+DEPENDENCIES = ['insteon_plm']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the INSTEON PLM device class for the hass platform."""
+
+    state_list = []
+    plm = hass.data['insteon_plm']
+
+    for deviceInfo in discovery_info:
+        address = deviceInfo['address']
+        device = plm.devices[address]
+        stateKey = deviceInfo['stateKey']
+        newnames = deviceInfo['newnames']
+
+        state_list.append(InsteonPLMSensorDevice(hass,
+                                                 device,
+                                                 stateKey,
+                                                 newnames))
+
+    async_add_devices(state_list)
+
+
+class InsteonPLMSensorDevice(Entity):
+    """A Class for an Insteon device."""
+
+    def __init__(self, hass, device, stateKey, newnames):
+        """Initialize the binarysensor."""
+        self._hass = hass
+        self._state = device.states[stateKey]
+        self._device = device
+        self._newnames = newnames
```

Comment:
Remove newnames feature

Reply:
Given that we have a plan for this in the future, would prefer to keep it in.